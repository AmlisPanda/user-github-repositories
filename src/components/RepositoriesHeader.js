import React from 'react';

export class RepositoriesHeader extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="reposHeader">
                <h2>Repositories</h2>

                <div>Found {this.props.reposNb} repositories. Showing top {this.props.top}.</div>

                <select id="sortList" onChange={this.props.onSort}>
                    <option value="0">Most recent first</option>
                    <option value="1">Oldest first</option>
                </select>
            </div>
        )
    }
}
