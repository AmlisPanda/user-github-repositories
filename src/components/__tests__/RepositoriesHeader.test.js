import React from 'react';
import renderer from 'react-test-renderer';
import { RepositoriesHeader } from '../RepositoriesHeader';

describe('RepositoriesHeader', () => {
    it('renders the component', () => {
        const tree = renderer.create(<RepositoriesHeader />).toJSON();
        expect(tree).toMatchSnapshot();
    });

})
