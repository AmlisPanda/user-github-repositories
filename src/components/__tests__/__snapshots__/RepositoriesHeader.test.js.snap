// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoriesHeader renders the component 1`] = `
<div
  className="reposHeader"
>
  <h2>
    Repositories
  </h2>
  <div>
    Found 
     repositories. Showing top 
    .
  </div>
  <select
    id="sortList"
    onChange={undefined}
  >
    <option
      value="0"
    >
      Most recent first
    </option>
    <option
      value="1"
    >
      Oldest first
    </option>
  </select>
</div>
`;
